### PR TITLE
Minor fixes to concensus relevant parameters

### PR DESCRIPTION
**Pull request by langerhans**
Three fixes:
* Always use simplified rewards for regtest.  
* Add changes from #1133 
* Check Coinbase maturity based on the 145k fork rules in all places.

### CODE REVIEW

File: src/auxpow.cpp [hunk 1]
```diff
@@ -89,7 +89,11 @@ int CMerkleTx::GetBlocksToMaturity() const
 {
     if (!IsCoinBase())
         return 0;
-    return std::max(0, (COINBASE_MATURITY + 1) - GetDepthInMainChain());
+    int nDepth = GetDepthInMainChain();
+    int nCoinbaseMaturity = nDepth < COINBASE_MATURITY_SWITCH
+        ? COINBASE_MATURITY_OLD
+        : COINBASE_MATURITY;
+    return std::max(0, (nCoinbaseMaturity + 1) - nDepth);
```
Comment (rnicoll) [inline src/auxpow.cpp]: Doesn't AuxPoW turn on only after the switch, so it's never the old value?

File: src/qt/transactiondesc.cpp [hunk 2]
```diff
@@ -264,7 +264,10 @@ QString TransactionDesc::toHTML(CWallet *wallet, CWalletTx &wtx, TransactionReco
 
     if (wtx.IsCoinBase())
     {
-        quint32 numBlocksToMaturity = COINBASE_MATURITY +  1;
+        quint32 nDepth = wtx.GetDepthInMainChain();
+        quint32 numBlocksToMaturity = nDepth < COINBASE_MATURITY_SWITCH
+            ? COINBASE_MATURITY_OLD + 1
+            : COINBASE_MATURITY + 1;
```
Comment (rnicoll) [inline src/qt/transactiondesc.cpp]: As we're well past the switch point, is this useful any more?

**Comment by rnicoll** [comment]
Wrote line comments then realised it applies all over the place - as we're now several hundred thousand blocks past the maturity switch, we only need it for allowing legacy transactions during initial sync, which I don't believe any of those methods are used for?

**Comment by langerhans** [comment]
Hm, right. I did that so I wouldn't have to mine 240 regtest blocks to mature coins in the RPC tests. But then I pulled in the change from #1133 and 240 blocks take a blink to mine now anyway. Pretty sure I can remove the other changes then. Will do later today :) Thanks for the hint!

**Comment by patricklodder** [comment]
I'd still like to put `COINBASE_MATURITY` in the chainparams though? 

I know that this will worsen compatibility with bitcoin but we actually forked this... so from a "Reference" p.o.v. it seems better if we put it in the chainparams where the other forks are described as well?

Any objection against that other than bitcoin compat?

**Comment by langerhans** [comment]
No objections from my side. I was thinking about doing it but wasn't quite sure about it cause I'd need to get the params everywhere. If I remove the check in all those additional places, I can move the maturity over too.

**Comment by rnicoll** [comment]
Sounds good to me, just didn't want to do it myself to be honest!

File: src/qt/transactiondesc.cpp [hunk 3]
```diff
@@ -264,8 +265,10 @@ QString TransactionDesc::toHTML(CWallet *wallet, CWalletTx &wtx, TransactionReco
 
     if (wtx.IsCoinBase())
     {
-        quint32 numBlocksToMaturity = COINBASE_MATURITY +  1;
-        strHTML += "<br>" + tr("Generated coins must mature %1 blocks before they can be spent. When you generated this block, it was broadcast to the network to be added to the block chain. If it fails to get into the chain, its state will change to \"not accepted\" and it won't be spendable. This may occasionally happen if another node generates a block within a few seconds of yours.").arg(QString::number(numBlocksToMaturity)) + "<br>";
+        quint32 nDepth = wtx.GetDepthInMainChain();
+        quint32 nCoinbaseMaturity = Params().GetConsensus(chainActive.Height()).nCoinbaseMaturity + 1;
+        LogPrintf("Depth: %i, Maturity: %i, Height: %i", nDepth, nCoinbaseMaturity, chainActive.Height());
```
Comment (rnicoll) [inline src/qt/transactiondesc.cpp]: Could you pull this LogPrintf out please

File: src/auxpow.cpp [hunk 4]
```diff
@@ -89,7 +90,9 @@ int CMerkleTx::GetBlocksToMaturity() const
 {
     if (!IsCoinBase())
         return 0;
-    return std::max(0, (COINBASE_MATURITY + 1) - GetDepthInMainChain());
+    int nDepth = GetDepthInMainChain();
```
Comment (rnicoll) [inline src/auxpow.cpp]: Does this need to be its own variable?

**Pull request merged**